### ISSUE TITLE
`no-self-argument` on Pydantic validator (decorator creates `classmethod`)

### ISSUE THREAD

**Issue by berzi** [PyCQA/pylint#6900]
### Bug description

```python
from pydantic import validator, BaseModel


class Test(BaseModel):
    a_field: int

    @validator
    def some_validator(cls, value):
        return value
```

This produces the error below, although pydantic's `validator` does indeed output a classmethod, and as such `cls` makes sense. See also [this issue on Pydantic](https://github.com/samuelcolvin/pydantic/issues/568) where it's ascertained that the issue lies ultimately with the linter.

The workarounds:
- Using `@classmethod` after `@validator`, though this doesn't seem to work for everyone, it produces warnings in IDEs such as pycharm, and is risky in general.
- Slapping `# pylint: disable=no-self-argument` on every line where a validator is used, but this is very unwieldy.

Since the code works and is correct, the linter should not complain about it.

Unfortunately this seems a rather complex issue for me to investigate since I never contributed to pylint before and don't have the free time to start now. Hopefully someone's willing to look into it.

### Command used

```shell
pylint src\temp.py
```

### Pylint output

```shell
************* Module temp
src\temp.py:8:4: E0213: Method should have "self" as first argument (no-self-argument)
```

### Expected behavior

No error, this is a known, if a bit unusual, pattern already recognised by editors such as PyCharm.

### Pylint version

```shell
pylint 2.14.1
astroid 2.11.5
Python 3.9.10 (tags/v3.9.10:f2f3f53, Jan 17 2022, 15:14:21) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 11

### Additional dependencies

pydantic==1.9.0

**Comment by guya-pecan** [comment]
+1

**Comment by YamchenkoYV** [comment]
Reproduced in version pydantic==1.9.1, pylint==2.4.2

**Comment by rationa1** [comment]
add version info
pydantic==1.9.0
pylint 2.14.4
astroid 2.11.6
Python 3.9.10 (main, Feb 22 2022, 13:54:07)
[GCC 11.2.0]

**Comment by mbyrnepr2** [comment]
Hi! I notice that there exists a plugin called `pylint-pydantic` [github](https://github.com/fcfangcc/pylint-pydantic) [PyPI](https://pypi.org/project/pylint-pydantic/#description)

The error reported in this issue is supressed by pylint-pydantic:
```python
pip install pylint-pydantic
pylint --load-plugins pylint_pydantic --extension-pkg-whitelist='pydantic' example.py
```

Would it be acceptable to defer to pylint-pydantic for this library-specific case, what do you think?

**Comment by berzi** [comment]
That is not a bad idea and thank you for finding it. I would still at least mention it in the documentation somewhere to make it more visibile (pydantic for example does the same to highlight plugins that silence false-positives related to pydantic code).

**Comment by Pierre-Sassoulas** [comment]
Could you confirm ``pylint-pydantic`` work for you @berzi ? Letting an external plugin handle that would be great but it has very few stars and adding it to our doc would drive a lot of traffic to it, we should make sure it's ready for it.

**Comment by berzi** [comment]
Sure. I'll make you know on Monday.

**Comment by berzi** [comment]
I can confirm that installing `pylint-pydantic` and enabling it:

```
# pyproject.toml
[tool.pylint.MASTER]
load-plugins = "pylint_pydantic"
```

Makes the error go away. :)

**Comment by Pierre-Sassoulas** [comment]
Thank you ! Let's close this issue once pylint-pydantic is documented on our side. We should also talk about other plugins at least about pylint-django at the same time.

**Comment by bcb** [comment]
Adding `@classmethod` also works.
